### PR TITLE
Random module documentation updates

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
Get random module docs up to date to reflect changes over the last two release cycles.

- [x] inspected docs

### CODE REVIEW

File: modules/standard/Random.chpl [hunk 1]
```diff
@@ -21,20 +21,27 @@
...
-    The ``randomStream`` type can be used to generate individual random numbers
-    via :proc:`randomStream.getNext` or to fill an array with random numbers via
-    :proc:`randomStream.fill`. There are also several other methods available
-    for manipulating arrays, parallel iteration, or adjusting the stream's
-    position.
+    This module provides the :record:`randomStream` type that represents a
+    conceptual stream of random numbers of a particular scalar type. Individual
+    numbers can be generated with :proc:`randomStream.next`, or an iterable
+    sequence of numbers can be generated by providing a domain argument to
+    ``next``. The stream's position in it's sequence can be updated with
```
Comment (benharsh) [inline modules/standard/Random.chpl:28]: I think it should be "in its sequence"

File: modules/standard/Random.chpl [hunk 2]
```diff
@@ -674,46 +661,56 @@ module Random {
     number of the specified type. This allows for the generation of random
     numbers in parallel, where each task involved in the parallel iteration can
     request random numbers within a particular range and traverse that range of
-    the sequence independently of other tasks (see :proc:`randomStream.iterate`
-    (*unstable*)).
+    the sequence independently of other tasks (see :proc:`randomStream.next`).
+
+    Although parallel iteration is supported, the type itself is not
+    thread-safe. In particular, it is not safe to call methods such as
+    :proc:`next` or :proc:`fill` on a ``randomStream`` from multiple tasks
```
Comment (benharsh) [inline modules/standard/Random.chpl:668]: "on the same ``randomStream``"

File: modules/standard/Random.chpl [hunk 3]
```diff
@@ -674,46 +661,56 @@ module Random {
...
-    tasks concurrently.
+    .. code-block:: chapel
 
-    Several high-level methods are provided to generate random numbers or
+      var A: [1..n] int;
+      var rs = new randomStream(int);
+
+      forall (r, a) in zip(rs.next(A.domain), A) do a = r;
+
+    * Create a random stream for each task:
```
Comment (benharsh) [inline modules/standard/Random.chpl:682]: Maybe: "... for each task using task-private variables" ?

File: modules/standard/Random.chpl [hunk 4]
```diff
@@ -674,46 +661,56 @@ module Random {
...
-        stream (*unstable*)
-      * :proc:`randomStream.getNth` to skip to a particular position in the
-        stream and return the value at that position (*unstable*)
-
-    A ``randomStream`` can be initialized with a user-provided seed value. Two
-    ``randomStream`` instances initialized with the same seed will produce
-    identical sequences of random numbers. When not provided explicitly, a seed
-    value will be generated in an implementation specific manner designed to
-    minimize the chance that two distinct instances of the ``randomStream``
-    will have the same seed.
```
Comment (benharsh) [inline modules/standard/Random.chpl:716]: Does this paragraph about a user-provided seed value need to be preserved at all? Is this behavior no longer relevant?

Comment (jeremiah-corrado) [reply modules/standard/Random.chpl:716]: The initializer docs and the module-level docs have the same information, so I figured this was safe to remove. lmk if you disagree.

**Pull request merged**